extension/encoding/awslogsencodingextension/internal/unmarshaler/s3-access-log/unmarshaler.go

```diff
@@ -223,22 +221,63 @@ func addField(field int, value string, resourceAttr *resourceAttributes, record
 			return fmt.Errorf("unexpected: request uri %q has no method", value)
 		}
 		record.Attributes().PutStr(semconv.AttributeHTTPRequestMethod, method)
+
 		path, remaining, _ := strings.Cut(remaining, " ")
 		if path == "" {
 			return fmt.Errorf("unexpected: request uri %q has no path", value)
 		}
-		record.Attributes().PutStr(semconv.AttributeURLPath, path)
-		scheme, remaining, _ := strings.Cut(remaining, " ")
-		if scheme == "" {
-			return fmt.Errorf("unexpected: request uri %q has no scheme", value)
+		res, err := url.Parse(path)
+		if err != nil {
+			return fmt.Errorf("request uri path is invalid: %w", err)
+		}
+		if res.Path != "" {
+			record.Attributes().PutStr(semconv.AttributeURLPath, res.Path)
+		}
+		if res.RawQuery != "" {
+			record.Attributes().PutStr(semconv.AttributeURLQuery, res.RawQuery)
+		}
+		if res.Scheme != "" {
+			record.Attributes().PutStr(semconv.AttributeURLScheme, res.Scheme)
+		}
+		if res.Fragment != "" {
+			record.Attributes().PutStr(semconv.AttributeURLFragment, res.Fragment)
+		}
+
+		protocol, remaining, _ := strings.Cut(remaining, " ")
+		if protocol == "" {
+			return fmt.Errorf("unexpected: request uri %q has no protocol", value)
 		}
 		if remaining != "" {
 			return fmt.Errorf(`request uri %q does not have expected format "<method> <path> <scheme>"`, value)
 		}
-		record.Attributes().PutStr(semconv.AttributeURLScheme, scheme)
+		name, version, err := netProtocol(protocol)
+		if err != nil {
+			return err
+		}
+		record.Attributes().PutStr(semconv.AttributeTLSProtocolName, name)
+		record.Attributes().PutStr(semconv.AttributeTLSProtocolVersion, version)
```

Comment:
should be `net.protocol.*`, not `tls.protocol.*`

Reply:
Of course, I have changed. So many fields :)